File: python/hot100/45. 二叉树的右视图.py
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root: return []
        q=deque([root])
        ans=[]
        while q:
            size=len(q)
            for i in range(size):
                node=q.popleft()
                if i==size-1:
                    ans.append(node.val)
                if node.left: q.append(node.left)
                if node.right: q.append(node.right)
        return ans

File: cpp/hot100/45. 二叉树的右视图.cpp
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if(!root) return {};
        vector<int> ans;
        queue<TreeNode*> q;
        q.push(root);
        TreeNode *p=root;
        int count=1;
        while(!q.empty()){
            int temp=0;
            while(count--){
                p=q.front();
                q.pop();
                if(p->left){
                    q.push(p->left);
                    temp++;
                }
                if(p->right){
                    q.push(p->right);
                    temp++;
                }
            }
            ans.push_back(p->val);
            count=temp;
        }
        return ans;
    }
};